synapse/handlers/auth.py

```diff
@@ -483,6 +483,11 @@ async def check_ui_auth(
 
         sid: Optional[str] = None
         authdict = clientdict.pop("auth", {})
+        if not isinstance(authdict, dict):
+            raise SynapseError(
+                400,
+                "Interactive auth not yet complete. Client data is not dictionary.",
+            )
```

Comment:
Hi @clokep @DMRobertson,  was some decision made on Pydantic? What should we do here for now?

If this function is called from every rest endpoint at the beginning should we keep it here, or move it upfront to rest layers which are using this.

Reply:
@prajjawal05 I think adding the validation would be a good step forward. I don't agree there's an open question. Today we're using pydantic 1.x and should not block good work because we might one-day update to pydantic 2.x.